### PR TITLE
[feature](Nereids) rewrite count(null) to 0

### PR DESCRIPTION
**Pull request by keanji-x**
## Proposed changes

select count(null) --> select 0

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/CountLiteralRewrite.java [hunk 1]
```diff
@@ -21,31 +21,59 @@
...
+                        // no need to rewrite
+                        return agg;
+                    }
+
+                    Map<Boolean, List<NamedExpression>> projectsAndAggFunc = newExprs.stream()
+                            .collect(Collectors.partitioningBy(Expression::isConstant));
+
+                    if (projectsAndAggFunc.get(false).isEmpty()) {
+                        // if there is no group by keys and other agg func, just return the one row Relations, such as
+                        // select count(null) from t
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/CountLiteralRewrite.java:60]: it is not right when base table is empty, think about
```sql
select 'abc' from t group by 'abc'
```

Comment (XieJiann) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/CountLiteralRewrite.java]: fixed

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/Expression.java [hunk 2]
```diff
@@ -223,6 +224,10 @@ public Expression withChildren(List<Expression> children) {
      * Whether the expression is a constant.
      */
     public boolean isConstant() {
+        if (this instanceof AggregateFunction) {
```
[the remark this answers is not in the document]
Comment (XieJiann) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/Expression.java:227]: Yes, it can be rewritten. I'll do it in next pr

**Comment by XieJiann** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by jackwener — APPROVED** [review]

**Review by jackwener — APPROVED** [review]

**Pull request merged**